src/Data/Detection/DetectionRepository.cs

```diff
@@ -17,14 +17,14 @@ public IEnumerable<MDetection> ByParameter() => _context.Set<EDetection>()
             .ToList()
             .Select(DetectionFactory.MakeModel);
 
-        public IEnumerable<MDetection> Insert(IEnumerable<MCreateDetection> input) => new List<MDetection>
-        {
-            new MDetection
+        public IEnumerable<MDetection> Insert(IEnumerable<MCreateDetection> input) => input
+            .Select(DetectionFactory.MakeEntity)
+            .Select(entity =>
             {
-                Class = "glass",
-                Score = 100,
-                Timestamp = DateTime.UtcNow,
-            },
-        };
+                _context.Add(entity);
+                _context.SaveChanges();
+                return entity;
+            })
```

Comment:
Should we extract this lambda into a separate private method?

Reply:
Should we also remove the unused import at the top of the file?

Reply:
Done